crates/ruff_linter/src/rules/pylint/rules/misplaced_bare_raise.rs

```diff
@@ -0,0 +1,100 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::statement_visitor::StatementVisitor;
+use ruff_python_ast::{statement_visitor, Stmt};
+use ruff_python_semantic::NodeRef;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+///
+/// This rule triggers an error when a bare raise statement is not in an except or finally block.
+///
+/// ## Why is this bad?
+///
+/// If raise statement is not in an except or finally block, there is no active exception to
+/// re-raise, so it will fail with a `RuntimeError` exception.
+///
+/// ## Example
+/// ```python
+/// def validate_positive(x):
+///     if x <= 0:
+///         raise
+/// ```
+///
+/// Use instead:
+/// ```python
+/// def validate_positive(x):
+///     if x <= 0:
+///         raise ValueError(f"{x} is not positive")
+/// ```
+#[violation]
+pub struct MisplacedBareRaise;
+
+impl Violation for MisplacedBareRaise {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("The raise statement is not inside an except clause")
+    }
+}
+
+#[derive(Default)]
+struct RaiseFinder<'a> {
+    raises: Vec<&'a Stmt>,
+}
+
+impl<'a, 'b> StatementVisitor<'b> for RaiseFinder<'a>
+where
+    'b: 'a,
+{
+    fn visit_stmt(&mut self, stmt: &'b Stmt) {
+        match stmt {
+            Stmt::Raise(_) => self.raises.push(stmt),
+            _ => statement_visitor::walk_stmt(self, stmt),
+        }
+    }
+}
+
+/// PLE0704
+pub(crate) fn misplaced_bare_raise(checker: &mut Checker, raise_stmt: &Stmt) {
+    for id in checker.semantic().current_statement_ids() {
+        let node = checker.semantic().node(id);
+        if let NodeRef::Stmt(stmt) = node {
+            match stmt {
+                Stmt::Try(try_stmt) => {
+                    // check whether the raise is in the except handlers
+                    let mut raise_finder = RaiseFinder::default();
+                    for handler in &try_stmt.handlers {
+                        raise_finder.visit_except_handler(handler);
+                    }
+                    if raise_finder.raises.contains(&raise_stmt) {
+                        return;
+                    }
+
+                    // if the raise is in the finally block, it might be ok if that finally is in
+                    // an except block
```

Comment:
Can you include a code example here in the comments to demonstrate the case that this is handling?

Reply:
Not needed anymore!